src/api/cvc4cpp.h

```diff
@@ -0,0 +1,4556 @@
+   * Parameters: 2
+   *   -[1]: ???
+   *   -[2]: ???
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  STRING_CHARAT,
+  /**
+   * String contains.
+   * Parameters: 2
+   *   -[1]: ???
+   *   -[2]: ???
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  STRING_STRCTN,
+  /**
+   * String index-of.
+   * Parameters: 3
+   *   -[1]: ???
+   *   -[2]: ???
+   *   -[3]: ???
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2, Term child3)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  STRING_STRIDOF,
+  /**
+   * String replace.
+   * Parameters: 3
+   *   -[1]: ???
+   *   -[2]: ???
+   *   -[3]: ???
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2, Term child3)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  STRING_STRREPL,
+  /**
+   * String prefix-of.
+   * Parameters: 2
+   *   -[1]: ???
+   *   -[2]: ???
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  STRING_PREFIX,
+  /**
+   * String suffix-of.
+   * Parameters: 2
+   *   -[1]: ???
+   *   -[2]: ???
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  STRING_SUFFIX,
+  /**
+   * Integer to string.
+   * Parameters: 1
+   *   -[1]: Term of sort Integer
+   * Create with:
+   *   mkTerm(Kind kind, Term child)
+   */
+  STRING_ITOS,
+  /**
+   * String to integer (total function).
+   * Parameters: 1
+   *   -[1]: Term of sort String
+   * Create with:
+   *   mkTerm(Kind kind, Term child)
+   */
+  STRING_STOI,
+  /**
+   * String of characters constant.
+   * Parameters:
+   *   See mkString()
+   * Create with:
+   *   mkString(const char* s)
+   *   mkString(const std::string& s)
+   *   mkString(const unsigned char c)
+   *   mkString(const std::vector<unsigned>& s)
+   *   mkConst(Kind kind, const char* s)
+   *   mkConst(Kind kind, const std::string& s)
+   */
+  CONST_STRING,
+  /**
+   * Conversion from string to regexp.
+   * Parameters: 1
+   *   -[1]: Term of sort String
+   * Create with:
+   *   mkTerm(Kind kind, Term child)
+   */
+  STRING_TO_REGEXP,
+  /**
+   * Regexp Concatenation .
+   * Parameters: 2
+   *   -[1]..[2]: Terms of Regexp sort
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  REGEXP_CONCAT,
+  /**
+   * Regexp union.
+   * Parameters: 2
+   *   -[1]..[2]: Terms of Regexp sort
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  REGEXP_UNION,
+  /**
+   * Regexp intersection.
+   * Parameters: 2
+   *   -[1]..[2]: Terms of Regexp sort

```

Comment:
Do you need this forward decl?

Reply:
What forward decl? Also, we already discussed forward decls today in the meeting, will get rid of them. But only after discussion is over.